vast_pipeline/utils/delete_run.py

```diff
@@ -0,0 +1,165 @@
+            _run_raw_sql(sql_cmd, cursor, log=False)
+
+            # Delete from related source
+            sql_cmd = f"DELETE FROM vast_pipeline_relatedsource WHERE from_source_id = {source_id};"
+            _run_raw_sql(sql_cmd, cursor, log=False)
+            sql_cmd = f"DELETE FROM vast_pipeline_relatedsource WHERE to_source_id = {source_id};"
+            _run_raw_sql(sql_cmd, cursor, log=False)
+
+            sql_cmd = f"DELETE FROM vast_pipeline_association WHERE source_id = {source_id};"
+            _run_raw_sql(sql_cmd, cursor, log=False)
+            
+            if i % 1000 == 0:
+                logger.info(f"Finished source id {source_id} ({i} of {n_source_ids})")
+        logger.info(f"Time to iterate over {n_source_ids} source ids: {timer.reset()} seconds")
+
+        # Delete source
+        sql_cmd = f"DELETE FROM vast_pipeline_source WHERE run_id = {p_run_id};"
+        _run_raw_sql(sql_cmd, cursor)
+        
+        # Enable triggers
+        #sql_cmd = "ALTER TABLE vast_pipeline_source ENABLE TRIGGER ALL;"
+        #_run_raw_sql(sql_cmd, cursor)
+
+        # Delete comments
+        sql_cmd = f"DELETE FROM vast_pipeline_comment WHERE object_id = {p_run_id};"
+        _run_raw_sql(sql_cmd, cursor)
+
+        # Fetch image IDs associated with the pipeline run
+        sql_cmd = f"SELECT image_id FROM vast_pipeline_image_run WHERE run_id = {p_run_id};"
+        _run_raw_sql(sql_cmd, cursor)
+        image_ids = cursor.fetchall()
+
+        # Iterate over each image ID and delete related information
+        n_image_ids = len(image_ids)
+        logger.info(f"Iterating over {n_image_ids} images to delete measurements and images")
+        timer.reset()
+
+        for image_id_tuple in image_ids:
+            image_id = image_id_tuple[0]
+            
+            # Check if the Image is associated with more than one run
+            sql_cmd = f"SELECT COUNT(*) FROM vast_pipeline_image_run WHERE image_id={image_id};"
+            _run_raw_sql(sql_cmd, cursor)
+            num_occurences = cursor.fetchone()[0]
+            
+            # Delete the link between the run and the image
+            sql_cmd = f"DELETE FROM vast_pipeline_image_run WHERE image_id = {image_id} AND run_id = {p_run_id};"
+            _run_raw_sql(sql_cmd, cursor, log=True)
+            
+            # If the image is associated with more than one run, do not delete the image.
+            if num_occurences > 1:
+                logger.debug("image_id %d is referenced by {num_occurences} other pipeline runs, not deleting", image_id)
+                continue
+
+            try:
+                sql_cmd = f"DELETE FROM vast_pipeline_measurement WHERE image_id = {image_id};"
+                _run_raw_sql(sql_cmd, cursor, log=True)
+            except Exception as e:
+                print("++++++++++++++++++++++++++++++++")
+                print(e, image_id)
+                print("++++++++++++++++++++++++++++++++")
```

Comment:
Rather use logger.error
```suggestion
                logger.error("%s %d", e, image_id)
```